go-controller/pkg/ovn/logical_switch_manager/logical_switch_manager_test.go

```diff
@@ -58,25 +77,88 @@ var _ = ginkgo.Describe("OVN Logical Switch Manager operations", func() {
 				}
 
-				// run the test for hybrid overlay enabled case
-				testHONode := testNodeSubnetData{
-					nodeName: "testNode2",
+				return nil
+			}
+			err := app.Run([]string{app.Name})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		})
+		ginkgo.It("creates IPAM for each subnet and reserves IPs correctly when HybridOverlay is enabled and address is passed", func() {
+			app.Action = func(ctx *cli.Context) error {
+				_, err := config.InitConfig(ctx, fexec, nil)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				testNode := testNodeSubnetData{
+					nodeName: "testNode1",
 					subnets: []string{
 						"10.1.1.0/24",
 						"2000::/64",
 					},
 				}
-				config.HybridOverlay.Enabled = true
-				expectedIPs = []string{"10.1.1.4", "2000::4"}
-				err = lsManager.AddNode(testHONode.nodeName, "", ovntest.MustParseIPNets(testNode.subnets...))
+				err = lsManager.AddNode(testNode.nodeName, "", ovntest.MustParseIPNets(testNode.subnets...))
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				allocatedHybridOverlayIFAddress, err := lsManager.AllocateHybridOverlay(testNode.nodeName, []string{"10.1.1.53"})
 				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				gomega.Expect(net.ParseIP("10.1.1.53").To4()).To(gomega.Equal(allocatedHybridOverlayIFAddress[0].IP))
+				gomega.Expect(true).To(gomega.Equal(lsManager.isAllocatedIP(testNode.nodeName, "10.1.1.53")))
 
-				ips, err = lsManager.AllocateNextIPs(testHONode.nodeName)
+				return nil
+			}
+			err := app.Run([]string{app.Name})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		})
+		ginkgo.It("creates IPAM for each subnet and reserves IPs correctly when HybridOverlay is enabled and no address is passed .3 avalible", func() {
+			app.Action = func(ctx *cli.Context) error {
+				_, err := config.InitConfig(ctx, fexec, nil)
 				gomega.Expect(err).NotTo(gomega.HaveOccurred())
-				for i, ip := range ips {
-					gomega.Expect(ip.IP.String()).To(gomega.Equal(expectedIPs[i]))
+
+				testNode := testNodeSubnetData{
+					nodeName: "testNode1",
+					subnets: []string{
+						"10.1.1.0/24",
+						"2000::/64",
+					},
 				}
 
+				err = lsManager.AddNode(testNode.nodeName, "", ovntest.MustParseIPNets(testNode.subnets...))
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				allocatedHybridOverlayIFAddress, err := lsManager.AllocateHybridOverlay(testNode.nodeName, []string{})
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				gomega.Expect(net.ParseIP("10.1.1.3").To4()).To(gomega.Equal(allocatedHybridOverlayIFAddress[0].IP))
+
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				gomega.Expect(true).To(gomega.Equal(lsManager.isAllocatedIP(testNode.nodeName, "10.1.1.3")))
+
+				return nil
+			}
+			err := app.Run([]string{app.Name})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		})
+		ginkgo.It("creates IPAM for each subnet and reserves IPs correctly when HybridOverlay is enabled and no address is passed .3 is not avalible", func() {
```

Comment:
"creates IPAM for each subnet and reserves a non-default IP address for hybrid overlay"

Reply:
done